packages/ui/Skinnable/useSkins.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import {determineSkin, determineVariants, getClassName} from './util';
+
+/**
+ * Allows a component to respond to skin changes via the Context API
+ *
+ * Example:
+ * ```
+ * <App skin="dark">
+ * 	<Section>
+ * 		<Button>Gray Button</Button>
+ * 	<Section>
+ * 	<Popup skin="light">
+ * 		<Button>White Button</Button>
+ * 	</Popup>
+ * </App>
+ * ```
+ *
+ * @class SkinContext
+ * @memberof ui/Skinnable
+ * @hoc
+ * @private
+ */
+const SkinContext = React.createContext(null);
+
+/**
+ * Configuration for `useSkins`
+ *
+ * @typedef {Object} useSkinsConfig
+ * @memberof ui/Skinnable
+ * @property {String}   defaultSkin     Default skin name when none has been specified or inherited.
+ * @property {String[]} defaultVariants Default variants when none have been specified or inherited.
+ * @property {String}   [skin]          Specific skin to apply to this instance which will take precedence over the default or the inherited value.
+ * @property {String[]} skins           List of allowed skins
+ * @property {String[]} [skinVariants]  Specific variants to apply to this instance which will take precedence over the default or the inherited value.
+ * @property {String[]} variants        List of allowed variants
```

Comment:
Similarly, is `variants` actually required?

Reply:
Nope